ts-rs/src/lib.rs

```diff
@@ -705,11 +676,15 @@ impl<I: TS> TS for Range<I> {
     where
         Self: 'static,
     {
-        ().push::<I>()
+        I::dependency_types()
     }
 
-    fn transparent() -> bool {
-        true
+    fn generics() -> impl TypeList
+    where
+        Self: 'static,
+    {
+        use std::marker::PhantomData;
+        ((PhantomData::<I>,), I::generics())
     }
 }
```

Comment:
~~So, this is equivalent to just returning `I::generics()`.  
I'm not sure that's correct, even though the tests pass. 
Shouldn't this return `().push::<I>().extend(I::generics())`? Right now, `I` itself would be missing from the TypeList, no?~~

Ahhh, I got confused by you constructing the TypeList manually here!  
Is there a reason you're not doing `().push::<I>().extend(I::generics())` or, even better, `I::generics().push::<I>`, instead?

Reply:
Not really, I just kinda got confused calling methods on `()` so I did it this way

Reply:
> I just kinda got confused calling methods on `()` so I did it this way

That tripped me up quite a bit in the macros, because the repetition syntax uses parentheses plus the unit type plus the calling of methods

Reply:
Hm, yeah, I understand.  
I do think it'd be a good idea to keep the implementation of `TypeList` private, and use the methods on it. That gives us the option to change TypeList is implemented however we like in the future.

In this particular case, `I::generics().push::<I>()` would probably be the nicest.  
To make the code more readable (and keep the implementation detail that the empty TypeList is `()` hidden), we could add a `const EMPTY_TYPELIST: () = ()` or `pub fn new() -> impl TypeList {}` to `typelist.rs`

For the macros, I try to use `quote!{}` or `quote![]` to disambiguate them, but yeah, there's still the `#()*` syntax.

In the end, this is not a big deal, if you've got strong opinions here. Personally, I prefer using the methods on the type.

Edit: I see that you already changed it, thanks! 😆 

Being able to call methods on `TypeList` was the motivation to make `dependency_types` a function in the first place. It could have been an associated type (e.g `type Dependencies = ((A), B), C::Dependencies)`), but I liked the idea of being able to pretend that `TypeList` is just a `Vec`.

Reply:
All of that makes a lot of sense! I'll open another PR to apply this change to the impls I changed in #241, all of which manually build a TypeList